apis/ec2/src/main/java/org/jclouds/ec2/compute/domain/EC2HardwareBuilder.java

```diff
@@ -139,6 +135,20 @@ public EC2HardwareBuilder virtualizationType(VirtualizationType virtualizationTy
       return this;
    }
 
+   public EC2HardwareBuilder virtualizationTypes(VirtualizationType ...virtualizationTypes) {
+      Preconditions.checkArgument(virtualizationTypes.length > 0, "At least one virtualization type is required.");
+      if (virtualizationTypes.length == 1) {
+         this.virtualizationType = new RequiresVirtualizationType(virtualizationTypes[0]);
+      } else {
+         List<RequiresVirtualizationType> supportedVirtualizationTypes = Lists.newArrayList();
```

Comment:
[minor] Use `Iterables.transform` to lazily create the entities and only create the ones being considered.